### PR TITLE
Read files as UTF8 or binary explicitly

### PR DESCRIPTION
**Pull request by janpio**
(as this makes a difference on Windows)

### CODE REVIEW

File: spaceship/spec/portal/portal_stubbing.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
 class PortalStubbing
   class << self
     def adp_read_fixture_file(filename)
-      File.read(File.join('spaceship', 'spec', 'portal', 'fixtures', filename))
```
Comment (lacostej) [inline spaceship/spec/portal/portal_stubbing.rb]: I am not that fond of the hardcoded file name in the `adp_read_fixture_file` implementation. Alternatives are maybe to have
* a different function for bin reading ( `adp_binread_fixture_file`)
* an explicit parameter (`adp_read_fixture_file(file, binary: false)`)

**Review by lacostej — COMMENTED** [review]
There are many more places where we call File.read without specifying the encoding. Will they create an issue?

**Comment by janpio** [comment]
> There are many more places where we call File.read without specifying the encoding. Will they create an issue?

That is possible and also very probable - it will pop up when we actually start using fastlane on Windows and testing the individual tools and actions.

My hope for the long term is there is a better solution than to provide `encoding: "utf-8"` for each `File.read` that needs it (like configuring a system wide default or similar), but for the short term this was the easiest/simplest way to get the tests to pass. So this is definitely an issue that will be tackled later on.

(If it will not appear during normal fastlane usage, then I think it is good I didn't invest more time into finding that "bigger" solution. But I honestly don't expect that to happen.)

File: spaceship/spec/portal/portal_stubbing.rb (hunk 1 above)
Comment (janpio) [reply spaceship/spec/portal/portal_stubbing.rb]: Yes, that is one ugly piece of code.

I like the idea of two different functions (param is not obvious enough imho) and will implement this.

Comment (janpio) [reply spaceship/spec/portal/portal_stubbing.rb]: This is now fixed.

File: spaceship/spec/portal/portal_stubbing.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+      File.binread(File.join('spaceship', 'spec', 'portal', 'fixtures', filename))
```
Comment (lacostej) [inline spaceship/spec/portal/portal_stubbing.rb]: both are binread now?

**Review by lacostej — CHANGES_REQUESTED** [review]

File: spaceship/spec/portal/portal_stubbing.rb (hunk 2 above)
Comment (janpio) [reply spaceship/spec/portal/portal_stubbing.rb]: Ahem, while this of course works this was not the goal. 
That's why we have review - sorry about that.
Fixed.

**Review by lacostej — APPROVED** [review]
great

**Pull request merged**